hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/types/HazelcastObjectType.java

```diff
@@ -109,13 +114,29 @@ private static void generateFullTypeString(HazelcastObjectType type, StringBuild
         sb.append(')');
     }
 
+    private static void escape(StringBuilder sb, String value) {
+        for (char c : value.toCharArray()) {
+            if (c == '(' || c == ':' || c == ',' || c == ')') {
```

Comment:
shouldn’t we also escape backslash?

Reply:
In this version (of `QueryDataType` and `HazelcastObjectType`), there cannot be two different types whose digest strings are the same, so we don't need to escape backslash. In fact, even a strict subset of the current metacharacters can give the same guarantee, but I didn't want to complicate the implementation.